### ISSUE TITLE
Add bad_link and image_sample autotags for H@H urls

### ISSUE THREAD

**Issue by hdk5** [danbooru/danbooru#5508]
e-hentai's H@H has at least two different direct image url formats:

`https:\/\/[\w.]+\.hath\.network(?::\d+)?\/h\/(?<image_hash>\w+)-.+;xres=(?<sample_size>\d+|org)\/(?<file_name>.+)`
(simply `https://{...}.hath.network/h/{image_hash}-{...};xres={sample_size}/{file_name}`)
If `sample_size` is not `org`, the image is resampled

`https:\/\/[\w.]+\.hath\.network(?::\d+)?\/om\/\d+\/(?<image_hash>\w+)-.+\/(?<sample_size>\d+)\/\w+\/(?<file_name>.+)`
(simply `https://{...}.hath.network/om/{...}/{image_hash}-{...}/{sample_size}/{...}/{filename}`)
If `sample_size` is not `0`, the image is resampled

On a side note: original gallery (but not exact page) url can be found by image hash using `https://exhentai.org/?f_shash=`.

**Comment by nonamethanks** [comment]
Reopening because that pull didn't actually add sample recognition for e-hentai, just the framework to support it.

**Comment by hdk5** [comment]
No, it actually did https://github.com/danbooru/danbooru/pull/5525/files#diff-f341e017b8beba6a619930d61b8eb0b6103888c8c0232e87512882ff6bcdd8ee

**Comment by nonamethanks** [comment]
Derp, I ran the tests but didn't realize it was under null.
